envoy/config/extension_config_provider.h

```diff
@@ -35,17 +35,17 @@ template <class Factory, class FactoryCallback> class ExtensionConfigProvider {
   virtual absl::optional<FactoryCallback> config() PURE;
 };
 
-template <class FactoryCallback> class DynamicExtensionConfigProviderBase {
+class DynamicExtensionConfigProviderBase {
 public:
   virtual ~DynamicExtensionConfigProviderBase() = default;
 
   /**
    * Update the provider with a new configuration.
-   * @param config is an extension factory callback to replace the existing configuration.
+   * @param proto_config is a candidate configuration update.
```

Comment:
What does candidate mean?

Reply:
Admittedly, I got this language from Kuat's PR, and interpreted it as not being applied yet. I can change it to just "is the new configuration"